### DISCUSSION TITLE
Copilot seems to have anomalous complementary results that are exploited by the grey industry?

### DISCUSSION THREAD

**Discussion by z2665** [community/community#59723] [Copilot Conversations] [answered]
### Select Topic Area

Bug

### Body

```
input 乐天
```
![image](https://github.com/community/community/assets/11162926/4a4a655d-7be0-4108-b4e9-e8aedc504bac)
The code completion results in the name of a gambling website, which seems to be an unreasonable completion result, is code completion being maliciously used as an advertisement, an attack?

**Comment by Akash1134** [comment, marked as answer]
Hi @z2665 ,

Because Codex, the model powering GitHub Copilot, was trained on publicly available code, its training set included personal data that was included in that code. From our internal testing, we found it to be very rare that GitHub Copilot suggestions included personal data verbatim from the training set. In your case, the output with username is also such example.

In some cases, the model might suggest what appears to be personal data – email addresses, phone numbers, GitHub usernames, etc. – but those suggestions are actually fictitious information synthesized from patterns in training data and therefore do not relate to any particular individual. For example, when one of our engineers prompted GitHub Copilot with, “My name is Mona and my birthdate is,” GitHub Copilot suggested a random, fictitious date of “December 12,” which is not Mona’s actual birthdate.

We have also implemented a filter that blocks emails when shown in standard formats, but it’s still possible to get the model to suggest this sort of content if you try hard enough. We will keep improving the filter system to be more intelligent to detect and remove more personal data from the GitHub Copilot suggestions.

In the future, if you get to see personal info being shared in snippets, please report incidents (e.g., offensive output, code vulnerabilities, apparent personal information in code generation) directly to [copilot-safety@github.com](mailto:copilot-safety@github.com) so that we can improve our safeguards. GitHub takes safety and security very seriously and we are committed to continually improving.

Thanks!

**Comment by z2665** [comment]
@Akash1134 But this complementary information I obtained is an actual existing gambling site that can be searched in Google. So I'm worried if it's some kind of ad attack using AI
